### PR TITLE
bpo-45847: Port nis module to PY_STDLIB_MOD (GH-29699)

### PR DESCRIPTION
**Pull request by tiran**
Signed-off-by: Christian Heimes <christian@python.org>

https://bugs.python.org/issue45847

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -3191,6 +3191,26 @@ if test "$have_glibc_memmove_bug" = yes; then
     AS_VAR_APPEND([LIBMPDEC_CFLAGS], [" -U_FORTIFY_SOURCE"])
 fi
 
+
+dnl check for NIS / libnsl dependencies
+dnl libnsl dependencies include tirpc includes and lib
+PKG_CHECK_MODULES([LIBNSL], [libnsl], [have_nis=yes], [
+  LIBNSL_CFLAGS=
+  AC_SEARCH_LIBS([yp_match], [nsl], [have_nis=yes], [have_nis=no])
```
Comment (erlend-aasland) [inline configure.ac]: Wrap with `WITH_SAVE_ENV`?

File: Modules/Setup.stdlib.in [hunk 2]
```diff
@@ -40,7 +40,7 @@
 @MODULE__QUEUE_TRUE@_queue _queuemodule.c
 @MODULE__RANDOM_TRUE@_random _randommodule.c
 @MODULE__TYPING_TRUE@_typing _typingmodule.c
-@MODULE__XXSUBINTERPRETERS_TRUE@_xxsubinterpreters __xxsubinterpretersmodule.c
+@MODULE__XXSUBINTERPRETERS_TRUE@_xxsubinterpreters _xxsubinterpretersmodule.c
```
Comment (tiran) [inline Modules/Setup.stdlib.in:43]: @erlend-aasland I'm sneaking in this typo fix, too.

File: configure.ac [hunk 3]
```diff
@@ -6166,6 +6188,9 @@ PY_STDLIB_MOD([_sha3], [test "$with_builtin_sha3" = yes])
 PY_STDLIB_MOD([_blake2], [test "$with_builtin_blake2" = yes])
 
 PY_STDLIB_MOD([_decimal], [], [], [$LIBMPDEC_CFLAGS], [$LIBMPDEC_LDFLAGS])
+PY_STDLIB_MOD([nis],
+  [], [test "$have_nis" = yes -a "$ac_cv_header_rpc_rpc_h" = yes],
```
Comment (erlend-aasland) [inline configure.ac:6192]: AFAICS, header check for `rpc/rpc.h` is missing.

**Review by erlend-aasland — CHANGES_REQUESTED** [review]

File: configure.ac (hunk 3 above)
Comment (tiran) [reply configure.ac:6192]: It's right there:

```
AS_VAR_IF([have_nis], [yes], [
  WITH_SAVE_ENV([
    CPPFLAGS="$LIBNSL_CFLAGS $CFLAGS"
    AC_CHECK_HEADERS([rpc/rpc.h])
  ])
])
```

Comment (erlend-aasland) [reply configure.ac:6192]: 🤦🏻

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**